blockchain/blockchain.go

```diff
@@ -1136,7 +1136,7 @@ func (bc *BlockChain) writeBlockWithStateSerial(block *types.Block, receipts []*
 	bc.mu.Lock()
 	defer bc.mu.Unlock()
 
-	if bc.HasBlockAndState(block.Hash(), block.NumberU64()) {
+	if bc.HasBlockAndState(block.Hash(), block.NumberU64()) && bc.CurrentBlock().NumberU64() >= block.NumberU64() {
```

Comment:
How about creating a new function for this? Something like `CanInsertBlock()`?

Reply:
`ShouldTryInserting`?

Reply:
Good. Good. Very good. I'll reflect this comment.